lib/chef/provider/package/apt.rb

```diff
@@ -124,11 +124,6 @@ def unlock_package(name, version)
 
         private
 
-        # @return [String] package name with or without anchors attached to it.
-        def resolve_package(pkg)
-          new_resource.anchor_package_regex ? "^#{pkg}$" : pkg
-        end
-
```

Comment:
@thheinen This looks like a regression. This was done to fix a bug fyi https://github.com/chef/chef/pull/13873